### ISSUE TITLE
validatePatch is not removing the required field

### ISSUE THREAD

**Issue by gsavoie** [feathers-plus/generator-feathers-plus#165]
### Steps to reproduce

I noticed this code was generated for a `validatePatch`

```
let patch = merge({},
  base,
  { required: undefined },
  // !code: patch_more // !end
);
```
"Removing" the `required` field makes sense to me considering you are not sending all fields over when doing a patch.

The problem is that this override doesn't work with `lodash.merge`. From what I saw the generator was using `deepMerge` before, which was working this way.

Here's a runkit showing the difference https://runkit.com/embed/dz312rk9eo5w

Can't really suggest an alternative as I'm not too sure what was the reason to move to `lodash.merge` in the first place.

### Expected behavior
`validatePatch` should ignore the `required` field from the schema.

### Actual behavior
`validatePatch` does validate the `required` field.

**Comment by eddyystop** [comment]
deepMerge was rfemoved as I ran into a problem with it. I don't remember what it was. How does lodash.merge (mis-) handle this situation?

**Comment by gsavoie** [comment]
Apparently, it could be possible to do it with `lodash.mergeWith` which is the same thing but allows a customizer parameter

**Comment by eddyystop** [comment]
Resolved via https://github.com/feathers-plus/generator-feathers-plus/pull/170

Published as f+/cli v0.7.76 using generator v0.6.74
